diff --git a/node_modules/expo-media-library/android/build.gradle b/node_modules/expo-media-library/android/build.gradle
index cbd1780..4ca0cd9 100644
--- a/node_modules/expo-media-library/android/build.gradle
+++ b/node_modules/expo-media-library/android/build.gradle
@@ -79,7 +79,7 @@ dependencies {
 
   testImplementation "org.robolectric:robolectric:4.3.1"
 
-  api "androidx.exifinterface:exifinterface:1.0.0"
+  api "androidx.exifinterface:exifinterface:1.3.0"
 
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${safeExtGet('kotlinVersion', '1.4.21')}"
 }
